snappy_wrappers/wrappers/somatic_variants_qc/summarize-vcf.py

```diff
@@ -0,0 +1,269 @@
+
+
+# Adding A>C G>T grouping
+# Table for VAF
+# Genotype check > report different genotypes
+# Checking number multi allelic
+# Support alternative allel in the normal
+# #alt in normal + average ALT + max ALT
+# snappy pipeline for this
+
+##### SNAPPY PIPELINE
+## WRAPPER
+# - Checking the configuration file
+# - Checking input file
+## PIPELINE
+# - __init__ :
+
+CONTIG = ["chr" + str(i) for i in range(1, 22)]
+CONTIG.append("chrX")
+CONTIG.append("chrY")
+
+
+def check_variant_in_bed(v_chrom, v_start, v_end, bed_intervals, padding=0):
+    # print(f'variant chrom: {v_chrom} - start: {v_start} - end:{v_end}')
+    record = bed_intervals.query(v_chrom, v_start - padding, v_end + padding)
+    if record:
+        return True
+    else:
+        return False
+
+
+# def calculate_exoms_length(bed_intervals):
+#     contigs= ["chr" + str(i) for i in range(1, 22)]
+#     contigs.append("chrX")
+#     contigs.append("chrY")
+#     length = 0
+#     for contig in contigs:
+#         records = bed_intervals.query(contig,0,250000000)
+#         for record in records:
+#             length = length + int(record[2]) - int(record[1])
+#     return length
+
+
+def get_variant_type(ref, alt):
+    if (len(alt) == 1) and (len(ref) == len(alt)):
+        return "snp"
+    elif len(alt) != len(ref):
+        return "indels"
+
+
+def check_sp_read(variant):
+    if get_variant_type(variant.REF, variant.ALT[0]) == "snp":
+        dp = variant.INFO["DP"]
+        if dp == 1:
+            return "minimal"
+        elif (dp > 1) and (dp <= 5):
+            return "limited"
+        if dp >= 10:
+            return "strong"
+
+
+def assign_class_snvs(variant, mt_mat):
+    temp = str(variant.REF) + ">" + str(variant.ALT[0])
+    # Following result from plot-vcfstats with bcftools
+    match temp:
+        case "A>C" | "C>A":
+            mt_mat[0] += 1
+        case "A>G" | "G>A":
+            mt_mat[1] += 1
+        case "A>T" | "T>A":
+            mt_mat[2] += 1
+        case "C>G" | "G>C":
+            mt_mat[3] += 1
+        case "C>T" | "T>C":
+            mt_mat[4] += 1
+        case "G>T" | "T>G":
+            mt_mat[5] += 1
+    return mt_mat
+
+
+def process_vcf_file(vcf_file, bed_file="", filter_file=False, padding=0):
+    total_v_count = 0
+    if filter_file:
+        # VAF
+        vaf = []
+        # numb of variants in and out of exoms
+        v_inside_exom = 0
+        v_outside_exom = 0
+        # numb of snps and indels
+        n_snps = 0
+        n_indels = 0
+        indels_length = []
+        # support read informations
+        minimal_rp_snvs_exom = 0
+        minimal_rp_snvs_nexom = 0
+        limited_rp_snvs_exom = 0
+        limited_rp_snvs_nexom = 0
+        strong_rp_snvs_exom = 0
+        strong_rp_snvs_nexom = 0
+        # mutation classes
+        mt_classes = [0] * 6
+        for variant in vcf_file:
+            if variant.CHROM in CONTIG:
```

Comment:
I don't think you should reduce the variants to those in the CONTIGs. The exclusion of variants on decoys or viral sequences should have happened during calling already. If you want to offer the user the possibility of another round of filtration, you could perhaps use something like the `ignore_chroms` mechanism used in other steps.